tests/auth_tests/test_forms.py

```diff
@@ -377,6 +378,24 @@ def test_unicode_username(self):
         self.assertTrue(form.is_valid())
         self.assertEqual(form.non_field_errors(), [])
 
+    @override_settings(AUTH_USER_MODEL='auth_tests.CustomUserWithLongerUsernameField')
+    def test_username_field_max_length_matches_user_model(self):
+        # We authenticate with a username of exactly 255 characters
+        # because we want to test that it is possible to authenticate
+        # with a username that is longer than the 254 default for the form
+        # (for custom user models that allow usernames of that length),
+        # but we cannot test username lengths longer than this due to
+        # VARCHAR limitations in MySQL
+        data = {
+            'username': 'fifteenchars000' * 17,  # 255 characters
+            'password': 'pwd',
+        }
+
+        CustomUserWithLongerUsernameField.objects.create_user(**data)
+        form = AuthenticationForm(None, data)
+        self.assertTrue(form.is_valid())
+        self.assertEqual(form.non_field_errors(), [])
```

Comment:
You're right, there is no point to this. I saw this check in a few other tests in the same module, so I tried to be consistent, but I agree that the world is a better place without it. 😛